service/src/main/java/greencity/service/ubs/CoordinateServiceImpl.java

```diff
@@ -0,0 +1,315 @@
+package greencity.service.ubs;
+
+import greencity.dto.CoordinatesDto;
+import greencity.dto.GroupedOrderDto;
+import greencity.dto.OrderDto;
+import greencity.entity.coords.Coordinates;
+import greencity.entity.order.Order;
+import greencity.exceptions.ActiveOrdersNotFoundException;
+import greencity.exceptions.IncorrectValueException;
+import greencity.repository.AddressRepository;
+import greencity.repository.OrderRepository;
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static greencity.constant.ErrorMessage.*;
+
+@org.springframework.stereotype.Service
+@RequiredArgsConstructor
+public class CoordinateServiceImpl implements CoordinateService {
+    private final AddressRepository addressRepository;
+    private final OrderRepository orderRepository;
+    private final ModelMapper modelMapper;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<GroupedOrderDto> getAllUndeliveredOrdersWithLiters() {
```

Comment:
`    @Override
    public List<GroupedOrderDto> getAllUndeliveredOrdersWithLiters() {
        Set<Coordinates> allCoords = addressRepository.undeliveredOrdersCoords();
        List<Order> allOrders = getAllUndeliveredOrders();
        List<GroupedOrderDto> allOrdersWithLitres = new ArrayList<>();
        for (Coordinates coordinates : allCoords) {
            int currentCoordinatesCapacity = addressRepository.capacity(coordinates.getLatitude(), coordinates.getLongitude());

            List<OrderDto> currentCoordinatesOrders = allOrders.stream()
                    .filter(o -> o.getUbsUser().getAddress().getCoordinates().equals(coordinates))
                    .map(o -> modelMapper.map(o, OrderDto.class))
                    .collect(Collectors.toList());

            allOrdersWithLitres.add(GroupedOrderDto.builder()
                    .amountOfLitres(currentCoordinatesCapacity)
                    .groupOfOrders(currentCoordinatesOrders)
                    .build());
        }
        return allOrdersWithLitres;
    }`